backend/src/main/java/com/example/subjecthub/controller/SubjectServiceController.java

```diff
@@ -3,27 +3,49 @@
 import com.example.subjecthub.Application;
 import com.example.subjecthub.api.SubjectServiceApi;
 import com.example.subjecthub.entity.Subject;
+import com.example.subjecthub.entity.Tag;
 import com.example.subjecthub.repository.SubjectRepository;
 import com.example.subjecthub.utils.FuzzyUtils;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.CrossOrigin;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.stream.Collectors;
 
 @RestController
 @ParametersAreNonnullByDefault
-@CrossOrigin(origins = "http://localhost:4200")
+@CrossOrigin(origins = "*")
 public class SubjectServiceController implements SubjectServiceApi {
 
     @Autowired
     private SubjectRepository subjectRepository;
 
     @Override
+    public Subject addTagToSubject(  @PathVariable Long subjectId, @RequestBody Tag tag) {
+
+        // gets the current subject
+        Subject current = subjectRepository.findOne(subjectId);
+
+        // adds the subject to the tag object (not sure if needed)
+        List<Subject> tagSubjects = new ArrayList<>();
+        tagSubjects.add(current);
+        tag.setSubjects(tagSubjects);
```

Comment:
We're not persisting the tag to the database if the tag does not yet exist. Firstly we should check if a tag for the same name already exists in the database. If it doesn't, we should save it. And then we can set the subjects.

Also, we're losing any tags that may exist for that subject already. You're creating a new List and saving overwriting that over the other list.